server/route.go

```diff
@@ -1,28 +1,92 @@
 package server
 
 import (
+	"errors"
 	"fmt"
 	"github.com/smancke/guble/protocol"
+	"runtime"
+	"sync"
+	"time"
+)
+
+var (
+	errEmptyQueue = errors.New("Empty queue")
+	errTimeout    = errors.New("Channel sending timeout")
+
+	defaultQueueCap = 50
 )
 
 // NewRoute creates a new route pointer
-func NewRoute(path, applicationID, userID string, channel chan *MessageForRoute) *Route {
-	return &Route{
+func NewRoute(path, applicationID, userID string, c chan *MessageForRoute) *Route {
+	route := &Route{
+		messagesC:     c,
+		queue:         newQueue(),
+		queueSize:     0,
```

Comment:
I don not really understand the concept of the queueSize, why is that dynamic?
Maybe it would be more clear, to use the capacity of the queue for that.